tests/language_specific/chinese/test_chinese_syntax.py

```diff
@@ -0,0 +1,392 @@
+    predicted_token_order = tuple(
+        node.token
+        for node in SIMPLE_CHINESE_DEPENDENCY_TREE_LINEARIZER.linearize(
+            DependencyTree(tree)
+        ).surface_token_order
+    )
+    assert predicted_token_order == ("wo3", "gei3", "ni3", "shu1")
+
+
+"""Tests Chinese adverbial modifiers, which typically occur before the verb 
+unless they are temporal modifiers"""
+
+
+def test_adverb_mods():
+    me = DependencyTreeToken("wo3", NOUN)
+    give = DependencyTreeToken("gei3", VERB)
+    you = DependencyTreeToken("ni3", NOUN)
+    book = DependencyTreeToken("shu1", NOUN)
+    slowly = DependencyTreeToken("man4 man", ADVERB)
+    tree = DiGraph()
+    tree.add_edge(me, give, role=NOMINAL_SUBJECT)
+    tree.add_edge(book, give, role=OBJECT)
+    tree.add_edge(you, give, role=INDIRECT_OBJECT)
+    tree.add_edge(slowly, give, role=ADVERBIAL_MODIFIER)
+    predicted_token_order = tuple(
+        node.token
+        for node in SIMPLE_CHINESE_DEPENDENCY_TREE_LINEARIZER.linearize(
+            DependencyTree(tree)
+        ).surface_token_order
+    )
+    assert predicted_token_order == ("wo3", "man4 man", "gei3", "ni3", "shu1")
+
+
+"""Tests Chinese prepositional phrase occurring pre-verb (indicates place where verbal action took place)"""
+
+
+def test_preverbial_prep():
+    me = DependencyTreeToken("wo3", NOUN)
+    at = DependencyTreeToken("dzai4", ADPOSITION)
+    table = DependencyTreeToken("jwo1 dz", NOUN)
+    on = DependencyTreeToken("shang4", NOUN)
+    eat = DependencyTreeToken("chr1", VERB)
+    tree = DiGraph()
+    tree.add_edge(at, table, role=CASE_SPATIAL)
+    tree.add_edge(on, table, role=NOMINAL_MODIFIER)
+    tree.add_edge(me, eat, role=NOMINAL_SUBJECT)
+    tree.add_edge(table, eat, role=OBLIQUE_NOMINAL)
+    predicted_token_order = tuple(
+        node.token
+        for node in SIMPLE_CHINESE_DEPENDENCY_TREE_LINEARIZER.linearize(
+            DependencyTree(tree)
+        ).surface_token_order
+    )
+    assert predicted_token_order == ("wo3", "dzai4", "jwo1 dz", "shang4", "chr1")
+
+
+"""Tests Chinese coverbial phrase occuring after the verb (indicating something about the action)"""
+
+
+def test_I_put_the_book_on_the_table():
+    me = DependencyTreeToken("wo3", NOUN)
+    book = DependencyTreeToken("shu1", NOUN)
+    ba = DependencyTreeToken("ba3", PARTICLE)
+    put = DependencyTreeToken("fang4", VERB)
+    at = DependencyTreeToken("dzai4", ADPOSITION)
+    table = DependencyTreeToken("jwo1 dz", NOUN)
+    on = DependencyTreeToken("shang4", NOUN)
+    tree = DiGraph()
+    tree.add_edge(ba, book, role=CASE_SPATIAL)
+    tree.add_edge(book, put, role=OBLIQUE_NOMINAL)
+    tree.add_edge(me, put, role=NOMINAL_SUBJECT)
+    tree.add_edge(at, table, role=CASE_SPATIAL)
+    tree.add_edge(on, table, role=NOMINAL_MODIFIER)
+    # TODO: this is a bit of a hack since I'm not sure this really counts as an IO, but I'm not sure what to classify it as
+    #  since PP's occur before the verb in Chinese
```

Comment:
We should open an issue to track this hack if we don't have one already.

Reply:
I just opened https://github.com/isi-vista/adam/issues/797